src/qibolab/instruments/bluefors.py

```diff
@@ -0,0 +1,70 @@
+import re
+import socket
+import threading
+from datetime import datetime
+
+from qibo.config import log
+
+
+class TemperatureController:
+    def __init__(self, IP_address):
+        self.IP_address = IP_address
+        self._thread = threading.Thread(target=self.get_data)
+        self._thread.daemon = (
+            True  # Daemonize the thread so it terminates when the main program exits
+        )
+        self._is_running = False
+        self._property = None
+        self._thread.start()
+
+    def get_data(self):
+
+        # Create a TCP/IP socket
+        client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+        # Connect the socket to the server
+        client_socket.connect((self.IP_address, 8888))
+        log.info("Bluefors Temperature Controller Connected")
+
+        self.data = {}
+        while True:
+            try:
+                # Receive data from the server
+                message = client_socket.recv(1024).decode()
+                # print(message)
+
+                # Define regular expression pattern to match the identifier, parameters, and values
+                pattern = r"([\w-]+): {'(\w+)':([\d\.]+), '(\w+)':([\d\.]+)}"
+
+                # Find all matches using regex
+                matches = re.findall(pattern, message)
+
+                # Extract identifier, parameters, and values
+                flange, _, temperature, _, timestamp = matches[0]
```

Comment:
```suggestion
                # Define regular expression pattern to match the identifier, parameters, and values
                pattern = r"([\w-]+): {'\w+':([\d\.]+), '\w+':([\d\.]+)}"

                # Find all matches using regex
                matches = re.findall(pattern, message)

                # Extract identifier, parameters, and values
                flange, temperature, timestamp = matches[0]
```

Since they are unused, you could also avoid the groups.